lib/plugins/aws/package/compile/events/sns/index.js

```diff
@@ -64,6 +64,9 @@ class AwsCompileSNSEvents {
                   if (topicArn.indexOf('arn:') === 0) {
                     const splitArn = topicArn.split(':');
                     topicName = splitArn[splitArn.length - 1];
+                    if (this.provider.options && splitArn[3] !== this.provider.options.region) {
```

Comment:
I'd propose to use `this.options.region`. This just requires to pass in the `options` as a second argument in the constructor and then set `this.options = options` in the constructor.

This is still "ugly" (due to the way the Framework deals with regions) but it's more robust compared to `this.provider`. We use this pattern (using `this.options.region`) in other plugins as well.